fedimintd/src/fedimintd.rs

```diff
@@ -464,6 +465,37 @@ impl Fedimintd {
     }
 }
 
+fn check_release_notes_ack() -> anyhow::Result<()> {
+    const VERSION: &str = "v0.4";
+    const FM_SKIP_REL_NOTES_ACK_ENV: &str = "FM_SKIP_REL_NOTES_ACK";
+    const FM_REL_NOTES_ACK_ENV: &str = "FM_REL_NOTES_ACK";
+    // the suffix here is to prevent people getting to smart and trying to
+    // automate-it-out change it for every release
+    const FM_REL_NOTES_CUR_VAL: &str = "0_4_xyz";
+
+    // This is here only for upgrade tests and other automated tests, where
+    // juggling different values might be impossible.
+    if is_env_var_set(FM_SKIP_REL_NOTES_ACK_ENV) {
+        warn!("Skipping release notes ack. This should never be happening in production");
+        return Ok(());
+    }
+
+    // The actual code-path we want end users to take
+    if std::env::var(FM_REL_NOTES_ACK_ENV).unwrap_or_default() == FM_REL_NOTES_CUR_VAL {
+        return Ok(());
+    }
+
+    eprintln!(
+        "This version of fedimintd has some criticallly important requirements you should be aware of."
+    );
+    eprintln!("Not following them will likely result in a consensus failure or other problems.");
+    eprintln!("To ensure you have read them, you must set specific environment variable to a specific value.");
```

Comment:
change to "Read release notes for instructions on how to perform the upgrade"

Reply:
@Kodylow What's the reasoning behind the change.

One thing I like to highlight is setting this env (and thus this message) is displayed irrespective of the fact of setting up or upgrading the Federation. So "how to perform the upgrade" might be confusing for the user that is not doing any upgrade.